src/bigquery.ts

```diff
@@ -2232,6 +2232,14 @@ export class BigQuery extends Service {
         job!.getQueryResults(options, callback as QueryRowsCallback);
         return;
       }
+      // If timeout override was provided, return error.
+      if (queryReq.timeoutMs) {
```

Comment:
Learning Leah - is what's happening here that if the user has passed a timeout, we make an error that gets called if necessary as part of the callback?

Additional question - does this not get handled as part of nodejs common? Context: as you know, I've messed around with retries/timeouts/etc in gax which handles all of that, so it's unusual for me to see a library having to make this error message

Reply:
This timeout is passed when calling `jobs.query`, so if the query doesn't run in the given time (in this code path we check if the job is not completed), then we throw an error that the query timed out. Is not handled by nodejs-common here because is a service level timeout, basically some calls on BQ accepts timeouts.